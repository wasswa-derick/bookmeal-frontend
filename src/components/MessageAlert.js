import React from "react";
import PropTypes from "prop-types";

export const getAlert = type => {
  switch (type) {
    case "success":
      return "alert alert-success";
    case "danger":
      return "alert alert-danger";
    case "warning":
      return "alert alert-warning";
    case "info":
    default:
      return "alert alert-info";
  }
};

const MessageAlert = ({ text, type }) => {
  const alertType = getAlert(type);
  return (
    <div className={alertType} role="alert">
      {text}
    </div>
  );
};

MessageAlert.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
};

export default MessageAlert;
