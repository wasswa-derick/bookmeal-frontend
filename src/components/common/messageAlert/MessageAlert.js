import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { renderToStaticMarkup } from "react-dom/server";
import SweetAlert from "sweetalert-react"; // eslint-disable-line import/no-extraneous-dependencies
import "sweetalert/dist/sweetalert.css"; // eslint-disable-line import/no-extraneous-dependencies
import { setMessage } from "../../../actions/message";

/**
 * @class MessageAlert
 * @extends {React.Component}
 */
class MessageAlert extends React.Component {
  close = () => {
    this.props.setMessage({
      text: "",
      show: false
    });
  };

  /**
   * @returns {HTMLElement} div
   * @memberof MessageAlert
   */
  render() {
    const { text, show } = this.props;
    return (
      <SweetAlert
        show={show}
        title="Book a Meal Notification"
        html
        text={renderToStaticMarkup(<p>{text}</p>)}
        onConfirm={this.close}
      />
    );
  }
}

MessageAlert.propTypes = {
  text: PropTypes.string.isRequired,
  show: PropTypes.bool.isRequired,
  setMessage: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  text: state.messageReducer.message.text,
  show: state.messageReducer.message.show
});

export default connect(mapStateToProps, { setMessage })(MessageAlert);
