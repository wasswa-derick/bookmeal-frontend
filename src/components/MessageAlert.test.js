import React from "react";
import { shallow } from "enzyme";
import MessageAlert from "./MessageAlert";

describe("Message Alert", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MessageAlert text="test msg" type="info" />);
  });

  it("should display correct message", () => {
    expect(wrapper.find("div").text()).toBe("test msg");
  });

  it("should render the right alert type", () => {
    expect(wrapper.find("div").hasClass("alert alert-info")).toBe(true);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
