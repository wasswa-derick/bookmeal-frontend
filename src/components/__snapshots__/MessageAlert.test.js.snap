// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Alert should render correctly 1`] = `
<div
  className="alert alert-info"
  role="alert"
>
  test msg
</div>
`;
